pkg/k8sutil/create.go

```diff
@@ -227,3 +232,100 @@ func createNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) erro
 func ListNamespaces(nsclient corev1typed.NamespaceInterface) (*v1.NamespaceList, error) {
 	return nsclient.List(context.TODO(), metav1.ListOptions{})
 }
+
+// ObjectMetadata uniquely identifies any object in the list of YAML manifests.
+type ObjectMetadata struct {
+	Version string
+	Kind    string
+	Name    string
+}
+
+// YAMLToUnstructured accepts a Kubernetes manifest in YAML format and returns an object of type
+// `unstructured.Unstructured`. This object has many methods that can be used by the consumer to
+// extract metadata from the Kubernetes manifest.
+func YAMLToUnstructured(yamlObj []byte) (*unstructured.Unstructured, error) {
+	u := &unstructured.Unstructured{}
+
+	var gvk *schema.GroupVersionKind
+
+	// Decode YAML into `unstructured.Unstructured`.
+	dec := yamlserializer.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	if _, _, err := dec.Decode(yamlObj, gvk, u); err != nil {
+		return nil, fmt.Errorf("decoding: %w", err)
+	}
+
+	return u, nil
+}
+
+// SplitYAMLDocuments separates a YAML file with multiple YAML documents separated by `---` into
+// separate YAML files and returns a list.
+func SplitYAMLDocuments(yamlObj string) ([]string, error) {
+	var ret []string
+
+	reader := yamlutil.NewYAMLReader(bufio.NewReader(strings.NewReader(yamlObj)))
+
+	for {
+		// Read the YAML document delimited by `---`.
+		yamlManifest, err := reader.Read()
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("reading the YAML: %w", err)
+		}
+
+		yamlManifest, err = removeYAMLComments(yamlManifest)
+		if err != nil {
+			return nil, fmt.Errorf("removing YAML comments: %w", err)
+		}
+
+		// Check if the YAML is empty.
+		if string(yamlManifest) == "null\n" {
+			continue
+		}
+
+		ret = append(ret, string(yamlManifest))
+	}
+
+	return ret, nil
+}
+
+// YAMLToObjectMetadata extracts Kubernetes metadata from any YAML manifest and return an
+// ObjectMetadata.
+func YAMLToObjectMetadata(yamlObj string) (ObjectMetadata, error) {
+	u, err := YAMLToUnstructured([]byte(yamlObj))
+	if err != nil {
+		return ObjectMetadata{}, fmt.Errorf("YAML to unstructured object: %w", err)
+	}
+
+	if u.GetAPIVersion() == "" {
+		return ObjectMetadata{}, fmt.Errorf("invalid configuration no 'APIVersion' specified:\n%s", yamlObj)
```

Comment:
Nit. Same to next error msg.

```suggestion
		return ObjectMetadata{}, fmt.Errorf("invalid configuration, no 'APIVersion' specified:\n%s", yamlObj)
```